csunplugged/topics/management/commands/_ProgrammingChallengesStructureLoader.py

```diff
@@ -32,79 +18,83 @@ def load(self):
             raise MissingRequiredFieldError(
                 self.structure_file_path,
                 ["lanugages", "difficulties"],
                 "Programming Challenge Structure"
             )
 
-        for (language, language_data) in languages.items():
+        # Add "-languages" to the structure filename
+        prog_languages_translation_filename = "{}-languages.yaml".format(
+            os.path.splitext(self.structure_filename)[0]
+        )
+        prog_languages_translations = self.get_yaml_translations(
+            prog_languages_translation_filename,
+            required_slugs=prog_languages.keys(),
+            required_fields=["name"]
+        )
+
+        for (prog_language, prog_language_data) in prog_languages.items():
 
-            if language_data is None:
+            if prog_language_data is None:
                 raise MissingRequiredFieldError(
                     self.structure_file_path,
-                    ["name", "number"],
+                    ["number"],
                     "Programming Challenge Language"
                 )
 
             # Check for required fields
-            language_name = language_data.get("name", None)
-            language_number = language_data.get("number", None)
-            if language_name is None or language_number is None:
+            prog_language_number = prog_language_data.get("number", None)
+            if prog_language_number is None:
                 raise MissingRequiredFieldError(
                     self.structure_file_path,
-                    ["name", "number"],
+                    ["number"],
                     "Programming Challenge Language"
                 )
 
             # Check if icon is given
-            if "icon" in language_data:
-                language_icon = language_data["icon"]
+            if "icon" in prog_language_data:
+                prog_language_icon = prog_language_data["icon"]
             else:
-                language_icon = None
+                prog_language_icon = None
 
-            new_language = ProgrammingChallengeLanguage(
-                slug=language,
-                name=language_name,
-                number=language_number,
-                icon=language_icon
+            new_prog_language = ProgrammingChallengeLanguage(
+                slug=prog_language,
+                number=prog_language_number,
+                icon=prog_language_icon
             )
 
-            new_language.save()
+            translations = prog_languages_translations.get(prog_language, dict())
+            self.populate_translations(new_prog_language, translations)
+            self.mark_translation_availability(new_prog_language, required_fields=["name"])
+            new_prog_language.save()
 
-            self.log("Added programming langauge: {}".format(new_language.__str__()))
+            self.log("Added programming langauge: {}".format(new_prog_language.__str__()))
```

Comment:
omg how have we not noticed this typo in the output before? 😆  `langauge` -> `language`

Reply:
Good spotting :laughing: